### PR TITLE
Add ares configurations

### PR DESCRIPTION
**Pull request by CatTanker**
Aimed at adding Ares Configurations to WAE's YR Config. 

Initial Commit consists of:

- Ares Trigger Events
- Ares Trigger Actions

Which have currently been tested through loading APRA2 campaign maps (which can now be opened without errors) and through comparing how FA2 and WAE save these to the map file. 

ToDo:
- [Scripts](https://ares-developers.github.io/Ares-docs/new/scripts.html)

Presently, I have included a [A] in front of any ares configurations, so that people using vanilla YR can tell that they will not work in the base game.

### CODE REVIEW

File: src/TSMapEditor/Config/Events.ini [hunk 1]
```diff
@@ -1,295 +1,463 @@
...
+Name = Under EM Pulse
+Description = [A] Triggers when the attached object is disabled by EMP
+
+[EMPulse2]
+Name = Under EM Pulse by House
+Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62
+P2Type=HouseType
+
+[EMPulse3]
+Name = Rmoved EM Pulse
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Name = Removed EM Pulse
```

File: src/TSMapEditor/Config/Events.ini [hunk 2]
```diff
@@ -1,295 +1,463 @@
...
+[TechnoTypeExists]
+Name=TechnoType Exists
+Description=Triggers when at least this many technos of the specified type exist on the map
+P1Type=-2
+P2Type=Number
+P3Type=Techno
+
+[TechnoTypeDoesNotExist]
+Name=TechnoType Does Not Exist
+Description=Triggers when there are no technos of the spercified type on the map
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description=Triggers when there are no technos of the specified type on the map

```

File: src/TSMapEditor/Config/Events.ini [hunk 3]
```diff
@@ -1,295 +1,463 @@
...
+P2Type=HouseType
+
+[DestroyedUnitsLand]
+Name=Destroyed Units, Land
+Description=Triggers when all land units of the specified house have been destroyed.
+P2Type=HouseType
+
+[BuildingDoesNotExist]
+Name=Building Does Not Exist
+Description=No buildings of the specified type (owned by the house of this trigger) exist on the map
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description=No buildings of the specified type (owned by the house of this trigger) exist on the map.
```

File: src/TSMapEditor/Config/Events.ini [hunk 4]
```diff
@@ -1,295 +1,463 @@
...
+P2Type=HouseType
+
+[EnteredOrOverflownBy]
+Name=Entered Or Overflown By
+Description=Triggers when objects owned by the house (of this trigger) move over the attached cell.
+P2Type=HouseType
+
+[TechnoTypeExists]
+Name=TechnoType Exists
+Description=Triggers when at least this many technos of the specified type exist on the map
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description=Triggers when at least this many technos of the specified type exist on the map.
```

File: src/TSMapEditor/Config/Events.ini [hunk 5]
```diff
@@ -1,295 +1,463 @@
...
+
+[TechnoTypeDoesNotExist]
+Name=TechnoType Does Not Exist
+Description=Triggers when there are no technos of the spercified type on the map
+P1Type=-2
+P3Type=Techno
+; ares stuff begins officially
+[EMPulse1]
+Name = Under EM Pulse
+Description = [A] Triggers when the attached object is disabled by EMP
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description = [A] Triggers when the attached object is disabled by EMP.
```

File: src/TSMapEditor/Config/Events.ini [hunk 6]
```diff
@@ -1,295 +1,463 @@
...
+
+[EMPulse2]
+Name = Under EM Pulse by House
+Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62.
```

File: src/TSMapEditor/Config/Events.ini [hunk 7]
```diff
@@ -1,295 +1,463 @@
...
+P2Type=HouseType
+
+[EMPulse3]
+Name = Rmoved EM Pulse
+Description = [A] Triggers when the attached object is reactivated by a “healing” EMP
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description = [A] Triggers when the attached object is reactivated by a “healing” EMP.
```

File: src/TSMapEditor/Config/Events.ini [hunk 8]
```diff
@@ -1,295 +1,463 @@
...
+
+[EMPulse4]
+Name = Rmoved EM Pulse by House
+Description = [A] Triggers when the specified house reactivates the attached object by a “healing” EMP. Triggers before 64
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description = [A] Triggers when the specified house reactivates the attached object by a “healing” EMP. Triggers before 64.
```

File: src/TSMapEditor/Config/Events.ini [hunk 9]
```diff
@@ -1,295 +1,463 @@
...
+Name = Under EM Pulse by House
+Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62
+P2Type=HouseType
+
+[EMPulse3]
+Name = Rmoved EM Pulse
+Description = [A] Triggers when the attached object is reactivated by a “healing” EMP
+
+[EMPulse4]
+Name = Rmoved EM Pulse by House
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Name = Removed EM Pulse by House
```

File: src/TSMapEditor/Config/Events.ini [hunk 10]
```diff
@@ -1,295 +1,463 @@
...
+Description = [A] Triggers when a driver or Vehicle Thief enters the attached object.
+
+[VehicleTakenHouse]
+Name = Vehicle Taken By House
+Description = [A] Triggers when a driver or Vehicle Thief of the specified house enters the attached object. Triggers before 69.
+P2Type=HouseType
+
+[Abducted]
+Name = Abducted
+Description = [A] Triggers when the attached object is abducted
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
Description = [A] Triggers when the attached object is abducted.
```

File: src/TSMapEditor/Config/Events.ini [hunk 11]
```diff
@@ -1,295 +1,463 @@
...
+P1Type=-2
+P2Type=Number
+P3Type=Techno
+
+[TechnoTypeDoesNotExist]
+Name=TechnoType Does Not Exist
+Description=Triggers when there are no technos of the spercified type on the map
+P1Type=-2
+P3Type=Techno
+; ares stuff begins officially
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion

```

File: src/TSMapEditor/Config/Events.ini [hunk 12]
```diff
@@ -1,295 +1,463 @@
...
+Name = All KeepAlives Destroyed
+Description = [A] Triggers when the specified house has no more KeepAlive=yes objects.
+P2Type = HouseType
+
+[KeepALivesDestroyed]
+Name = All KeepAlive Buildings Destroyed
+Description = [A] Triggers when the specified house has no more buildings with KeepAlive=yes.
+P2Type = HouseType
+
+;end of ares stuff, yay
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Events.ini]:
```suggestion
```

File: src/TSMapEditor/Config/Actions.ini [hunk 13]
```diff
@@ -1,619 +1,817 @@
...
+[RetintBlue]
+Name=Retint Blue
+Description=Retint the Blue portion of the screen, %
+P2Type=Number
+
+[JumpCameraHome]
+Name=Jump Camera Home
+Description=Just like hitting the 'H' key.
+
+;begin additional ares stuff
```
Comment (MortonPL) [inline src/TSMapEditor/Config/Actions.ini]:
```suggestion
```

**Review by MortonPL — CHANGES_REQUESTED** [review]
Found some typo's and missing punctuation. There are also issues with inconsistent placement of spaces for entries, i.e.:

```ini
[HouseDoesntOwnTechno]
Name =  House Doesn’t Own Techno Type
Description = [A] Triggers while the owning house has fewer than count instances of the specified techno type.
P1Type=-2
P2Type=Number
P3Type=Techno
```

**Comment by MortonPL** [comment]
I'm also not sure whether to use [A] or [Ares] or keep the `yr` branch for pure vanilla and have Ares and Phobos elsewhere.

File: src/TSMapEditor/Config/Events.ini (hunk 11 above)
Comment (ZivDero) [reply src/TSMapEditor/Config/Events.ini]: Maybe comments wouldn't hurt, but in a more serious tone
Along the lines of `; ********** ARES ***********`

**Comment by ZivDero** [comment]
> I'm also not sure whether to use [A] or [Ares] or keep the `yr` branch for pure vanilla and have Ares and Phobos elsewhere.

Desciptions don't need any markings imo, the name can have an [Ares]/[Phobos]
I think having separate configs for Ares/Phobos woud be more of a nuisance.

**Comment by ZivDero** [comment]
Also what's up with the diff containing the whole config being rewritten?

**Comment by CatTanker** [comment]
image
My main concern with `[Ares]` is that it takes a lot of space up in the already-existent event/actions list, meaning that in the more severe cases only the first work can be read, which is a limitation if you have 5+ events all beginning with `[Ares] Reverse...`
Ultimately these may have to be resized if we keep with `[Ares]`, and one day `[Phobos]` option.

No clue why the diff is broken.

**Comment by CatTanker** [comment]
I have tested each event/action/script from producing a test map inside WAE and comparing the results to E1's FA2sp's understanding. Fixed several errors and updated a few descriptions.  Not much more I can do here.

**Comment by MortonPL** [comment]
Can be closed in favor of #202.

**Pull request closed without merging**